### PR TITLE
CSS modules: form components

### PR DESCRIPTION
**Pull request by dzearing**
Note: this PR is for the `module-css-updates` branch. We'll merge all changes there, publish a beta build, test some projects with that, then merge into master and publish after we've done some internal testing.

To make the PR easier to read, you can disable whitespace on the diff by adding the ?w=1 parameter to the PR url:

https://github.com/OfficeDev/office-ui-fabric-react/pull/1049/files?w=1

Checkbox, ChoiceGroup, Icon, Image, ColorPicker, MarqueeSelection, Layer, Dropdown, TextField, Slider, Spinner, ProgressIndicator, and Rating all updated to use css modules.

### CODE REVIEW

File: packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.scss [hunk 1]
```diff
@@ -32,198 +32,195 @@ $ms-checkbox-highContrast-iconBoxSize: 16px;
...
+  position: relative;
+}
 
-  // The hidden input
-  .ms-Checkbox-input {
-    position: absolute;
-    opacity: 0;
-    top: 8px
-  }
+.textLabel {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.scss]: I'm confused on `textLabel` and `checkboxLabel`.

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx [hunk 2]
```diff
@@ -60,27 +61,27 @@ export class ChoiceGroup extends BaseComponent<IChoiceGroupProps, IChoiceGroupSt
       // Need to assign role application on containing div because JAWS doesnt call OnKeyDown without this role
       <div role='application' className={ className }>
         <div
-          className='ms-ChoiceFieldGroup'
+          className={ css('ms-ChoiceFieldGroup', styles.root) }
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx]: We can do more cleanup in the future, but we're inconsistent here. The component is `ChoiceGroup` and then we have classes like `ms-ChoiceFieldGroup` and `ms-ChoiceField`?

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx [hunk 3]
```diff
@@ -60,27 +61,27 @@ export class ChoiceGroup extends BaseComponent<IChoiceGroupProps, IChoiceGroupSt
...
 
           { options.map((option) => (
             <div
               key={ option.key }
-              className={ css('ms-ChoiceField', {
-                'ms-ChoiceField--image': !!option.imageSrc || !!option.iconProps,
-                'is-inFocus': option.key === keyFocused
+              className={ css('ms-ChoiceField', styles.choiceField, {
+                ['ms-ChoiceField--image ' + styles.imageChoiceField]: !!option.imageSrc || !!option.iconProps,
+                ['is-inFocus' + styles.isInFocus]: option.key === keyFocused
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx]: `styles.choiceFieldIsInFocus`

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx [hunk 4]
```diff
@@ -60,27 +61,27 @@ export class ChoiceGroup extends BaseComponent<IChoiceGroupProps, IChoiceGroupSt
...
+          ) }
 
           { options.map((option) => (
             <div
               key={ option.key }
-              className={ css('ms-ChoiceField', {
-                'ms-ChoiceField--image': !!option.imageSrc || !!option.iconProps,
-                'is-inFocus': option.key === keyFocused
+              className={ css('ms-ChoiceField', styles.choiceField, {
+                ['ms-ChoiceField--image ' + styles.imageChoiceField]: !!option.imageSrc || !!option.iconProps,
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx]: `styles.choiceFieldIsImage`

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx [hunk 5]
```diff
@@ -126,48 +127,63 @@ export class ChoiceGroup extends BaseComponent<IChoiceGroupProps, IChoiceGroupSt
       <label
         htmlFor={ this._id + '-' + option.key }
         className={ css({
-          'ms-ChoiceField-field--image': !!option.imageSrc || !!option.iconProps,
-          'ms-ChoiceField-field': !option.imageSrc && !option.iconProps,
-          'is-checked': option.key === keyChecked,
-          'is-disabled': isDisabled
+          ['ms-ChoiceField-field--image ' + styles.imageField]: !!option.imageSrc || !!option.iconProps,
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx]: `styles.choiceFieldIsImage`

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx [hunk 6]
```diff
@@ -126,48 +127,63 @@ export class ChoiceGroup extends BaseComponent<IChoiceGroupProps, IChoiceGroupSt
...
+          ['ms-ChoiceField-field ' + styles.field]: !option.imageSrc && !option.iconProps,
+          ['is-checked ' + styles.isChecked]: option.key === keyChecked,
+          ['is-disabled ' + styles.isDisabled]: isDisabled
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx]: `styles.choiceFieldIsChecked` and `styles.choiceFieldIsDisabled`.

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx [hunk 7]
```diff
@@ -126,48 +127,63 @@ export class ChoiceGroup extends BaseComponent<IChoiceGroupProps, IChoiceGroupSt
...
-          option.imageSrc
-            ? <div className='ms-ChoiceField-innerField'>
-              <div className={ css('ms-ChoiceField-imageWrapper', { 'is-hidden': option.key === keyChecked }) }>
+          option.imageSrc && (
+            <div className={ css('ms-ChoiceField-innerField', styles.innerField) }>
+              <div className={ css(
+                'ms-ChoiceField-imageWrapper',
+                styles.imageWrapper,
+                {
+                  ['is-hidden ' + styles.isHidden]: option.key === keyChecked
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx]: `styles.imageWrapperIsHidden`

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.scss [hunk 8]
```diff
@@ -1,108 +1,107 @@
...
+}
+
+.slider {
+  position: relative;
+  height: 20px;
+  margin-bottom: 5px;
+  border: 1px solid $ms-color-neutralLight;
+  box-sizing: border-box;
+
+  &.isHue {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.scss]: `.sliderIsHue`

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.scss [hunk 9]
```diff
@@ -1,108 +1,107 @@
...
   }
 
-  .ms-ColorPicker-light {
-    position: absolute;
-    left: 0;
-    right: 0;
-    top: 0;
-    bottom: 0;
-    background: linear-gradient(to right, white 0%, transparent 100%);
+  &.isAlpha {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.scss]: `.sliderIsAlpha`

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorSlider.tsx [hunk 10]
```diff
@@ -65,12 +66,12 @@ export class ColorSlider extends React.Component<IColorSliderProps, IColorSlider
...
         ref='root'
-        className={ css('ms-ColorPicker-slider', className, {
+        className={ css('ms-ColorPicker-slider', styles.slider, className, {
           'is-adjusting': isAdjusting
         }) }
         onMouseDown={ this._onMouseDown }>
-        <div className='ms-ColorPicker-sliderOverlay' style={ overlayStyle } />
-        <div className='ms-ColorPicker-thumb is-slider' style={ { left: currentPercentage + '%' } } />
+        <div className={ css('ms-ColorPicker-sliderOverlay', styles.sliderOverlay) } style={ overlayStyle } />
+        <div className={ css('ms-ColorPicker-thumb is-slider', styles.thumb, styles.isSlider) } style={ { left: currentPercentage + '%' } } />
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/ColorPicker/ColorSlider.tsx]: `styles.thumbIsSlider`

File: packages/office-ui-fabric-react/src/components/Image/Image.scss [hunk 11]
```diff
@@ -1,82 +1,80 @@
...
+  overflow: hidden;
+}
 
-  // Modifier: The image frame should fill its parent element.
-  .ms-Image--maximizeFrame {
-    height: 100%;
-    width: 100%;
-  }
+// Modifier: The image frame should fill its parent element.
+.maximizeFrame {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Image/Image.scss]: `.rootHasMaximizedFrame`

File: packages/office-ui-fabric-react/src/components/Image/Image.scss [hunk 12]
```diff
@@ -1,82 +1,80 @@
...
-    @include left(50%);
-    top: 50%;
-    // transform-translateX mixin is not used here because a transateY is also required.
-    @include LTR {
-      transform: translate(-50%,-50%);
-    }
-    @include RTL {
-      transform: translate(50%,-50%);
-    }
+.imageCenter,
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Image/Image.scss]: All of these modifiers should contain the root class they are modifying. E.g. `.rootIsImageCover`

File: packages/office-ui-fabric-react/src/components/Layer/Layer.scss [hunk 13]
```diff
@@ -1,15 +1,13 @@
...
-  .ms-Layer--fixed {
-    position: fixed;
-    z-index: 1000000;
-    top: 0;
-    left: 0;
-    width: 100vw;
-    height: 100vh;
-    visibility: hidden;
-  }
+.fixed {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Layer/Layer.scss]: `.rootIsFixed`

File: packages/office-ui-fabric-react/src/components/Rating/Rating.scss [hunk 14]
```diff
@@ -8,79 +8,76 @@ $Rating-disabledColor: $ms-color-neutralTertiaryAlt;
...
-      text-align: center;
+    text-align: center;
 
-      color: $Rating-defaultColor;
+    color: $Rating-defaultColor;
 
-      &.is-selected {
-        color: $Rating-selectedColor;
-      }
+    &.isSelected {
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/Rating/Rating.scss]: `.rootIsSelected`

File: packages/office-ui-fabric-react/src/components/TextField/TextField.tsx [hunk 15]
```diff
@@ -209,18 +214,18 @@ export class TextField extends BaseComponent<ITextFieldProps, ITextFieldState> i
...
   }
 
-  private get _textElementClassName(): string {
+  private _getTextElementClassName(): string {
     const errorMessage: string = this._errorMessage;
     let textFieldClassName: string;
 
     if (this.props.multiline && !this.props.resizable) {
-      textFieldClassName = 'ms-TextField-field ms-TextField-field--unresizable';
+      textFieldClassName = css('ms-TextField-field ms-TextField-field--unresizable', styles.field, styles.isUnresizable);
```
Comment (mikewheaton) [inline packages/office-ui-fabric-react/src/components/TextField/TextField.tsx]: `styles.fieldIsUnresizable`

File: packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.scss (hunk 1 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/Checkbox/Checkbox.scss]: Checkbox has a label for the checkbox, and a label that displays the text. I also was confused about this, but i'm not changing the DOM to make it more clear.

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx (hunk 2 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx]: It is absolutely wonky. Not only that but the root isn't the root...

**Comment by micahgodbolt** [comment]
Overflow: hidden is on ms-Image instead of ms-Icon-imageContainer

imgpsh_mobile_save

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx (hunk 5 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx]: not quite right. this is `choicefield-field--image`, not `choicefield--image`!

Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx]: fieldIsImage

File: packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx (hunk 6 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ChoiceGroup/ChoiceGroup.tsx]: nope; `fieldIsChecked`.

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.scss (hunk 8 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ColorPicker/ColorPicker.scss]: i'm starting to think this is inconsistent.

File: packages/office-ui-fabric-react/src/components/ColorPicker/ColorSlider.tsx (hunk 10 above)
Comment (dzearing) [inline packages/office-ui-fabric-react/src/components/ColorPicker/ColorSlider.tsx]: wait... I think this is unnecessary. this seems unnecessarily verbose. maybe this is a bad idea. You missed some above.

**Comment by micahgodbolt** [comment]
@dzearing this one has some conflicts on spinner

**Comment by micahgodbolt** [comment]
This is a duplicate of #1049. I pulled this branch down, fixed the conflicts and created a new PR.

**Pull request closed without merging**